src/OpenAL/Extensions/OpenTK.OpenAL.Extensions.EXT.Capture/Interfaces/ICaptureContext.cs

```diff
@@ -0,0 +1,59 @@
+
+using AdvancedDLSupport;
+using OpenTK.OpenAL.Native;
+using Device = System.Void;
+
+namespace OpenTK.OpenAL.Extensions.EXT.Capture
+{
+    /// <summary>
+    /// Defines the public API of capture-related functions in the Capture extension.
+    /// </summary>
+    [NativeSymbols(Prefix = "alc")]
+    public interface ICaptureContext
+    {
+        /// <summary>
+        /// Opens a capture device by name.
+        /// </summary>
+        /// <param name="deviceName">The name of the device.</param>
+        /// <param name="frequency">The frequency data should be captured at.</param>
+        /// <param name="format">The requested capture buffer format.</param>
+        /// <param name="size">The size of the capture buffer.</param>
+        /// <returns>The device pointer, or null.</returns>
+        unsafe Device* CaptureOpenDevice([CallerFree] string deviceName, uint frequency, BufferFormat format, int size);
```

Comment:
to be honest I think I'm against the usage of Device*(it does not even work for me). I think the usage of IntPtr would make more sense because if we want to make it a public API it cannot be used by e.g. VB.Net. But if it has another reason I think it should be as a secondary option at most, and I don't think the renaming is something you would normally do in C#(at least it does not seem to work that great, and I know I would like to be able to use it sometimes as well, as I'm a Cpp dev too)

Reply:
I've made some alterations since - the aliasing didn't work anyway. However, recall that this is meant to be a very thin wrapper around the OpenAL functions, and should match them as closely as possible. 

I have taken some liberties here and there, and those are certainly up for debate. Having an alternate "safe" overload that takes IntPtrs should be fine.

Reply:
What about struct that contain IntPtr as api + backed with unsafe